### DISCUSSION TITLE
Multiple aliased domains to point to one Nuxt project

### DISCUSSION THREAD

**Discussion by Marchiuzzz** [nuxt/nuxt#10459] [Questions (Nuxt 2)]
Hello,
I'm building a content management system, where the backend will be served by Laravel and frontend should be Nuxt.js. The key point of this system is that a bunch of domains (aliases, NOT sub-domains) would be pointing to the same project folder but would be serving different content from API based on visited domain. From what I've read, I can't figure out a way how it would be possible (if at all) to achieve this?

**Comment by gplusdotgr** [comment]
@Marchiuzzz this is a very generic question, but I'll try to elaborate :) 
since APIs can be configured to return different results based on different paths or endpoints, what blocks you from having some logic in your `asyncData()` hook of your page that makes this or that API call based on the domain? The url is accessible in the [Nuxt context](https://nuxtjs.org/docs/internals-glossary/context/) (route). You could have a Switch statement, or you could try to map your Vuex actions and use dynamic key names that accept the domain name. 

Sorry if the answer is not too specific.

**Reply by Marchiuzzz** [reply]
Sorry, I should've been more specific with my question. I do understand I can pass in current domain to the API over a request in the `fetch()` or `asyncData()`. I'm just wondering if the nature of Node.js server won't be causing problems while deploying such system as I don't quite understand how it's route matching works. For a php project - it's simple, I create a domain on my server and map the public folder to this domain. How does Node.js work? Will I be able to tell Node.js on which domain to host the project and then add different aliases for this domain (without redirecting) then they should all point to the same project but be accessible from different domains, correct?

**Reply by gplusdotgr** [reply]
Some setups I've seen use an additional nginx server receiving all traffic and doing the aliases and routing the traffic for you. Not sure if you have access to such functionality from inside Nuxt (Nuxt basically has an embedded node server serving it but not sure if you can customize it that much besides the config options)
check also https://nuxtjs.org/docs/configuration-glossary/configuration-server

**Reply by gplusdotgr** [reply]
update on the above; I think if you have multiple domains/subdomains that need to point to the same application, you probably need to configure this logic on your host/dns and not on the application layer

**Reply by Marchiuzzz** [reply]
You're right, this isn't configured inside nuxt.js but rather on the server

**Reply by gplusdotgr** [reply]
I hope I was able to help :)
